src/controllers/userController.ts

```diff
@@ -91,9 +91,13 @@ export class UserController {
   public async removeUser(req: Express.Request, res: Express.Response) {
     const { id } = <RemoveUserDto>matchedData(req);
     try {
-      const user = await userService.removeUser(id);
-      if (user) {
-        return res.status(StatusCodes.OK).send(user);
+      const userToRemove = await userService.removeUser(id);
+      if (userToRemove) {
+        if (req.user?.isAdmin || req.user?.id == id) {
+          return res.status(StatusCodes.OK).send(userToRemove);
+        } else {
+          return res.status(StatusCodes.FORBIDDEN).send(ReasonPhrases.FORBIDDEN);
+        }

```

Comment:
Do I understand this correctly? First you remove user, then you check if the user who sent the request to remove entity is admin or the user himself? If yes, then this is so wrong. What if I wasn't the user or admin and I sent a request to delete a user? It would delete the user and send me a response with 403.

Correct me if I'm wrong.

Reply:
What you should do (if I'm right) is to 

1. get the user by id and return 404 if user === null
2. check if user from request is authorised to delete entity, if not => 403
3. delete entity => 200 or 204, depending on what code you agreed on with Robert (both are correct, but if you use 200 in this delete endpoint, then use it in all delete endpoints, and if you want to use 204 there, you should use 204 in all delete endpoints)

Reply:
Whoops, yeah, I forgot to swap those lines